### PR TITLE
Refactor spacings and typography

### PR DESCRIPTION
**Pull request by florentmaitre**

### CODE REVIEW

File: theme-contract/src/main/java/com/orange/ods/theme/spacing/OdsSpacings.kt [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        override val extraSmall = OdsToken(OdsToken.Spacing.ExtraSmall, this@OdsSpacings.extraSmall)
+        override val small = OdsToken(OdsToken.Spacing.Small, this@OdsSpacings.small)
+        override val medium = OdsToken(OdsToken.Spacing.Medium, this@OdsSpacings.medium)
+        override val large = OdsToken(OdsToken.Spacing.Large, this@OdsSpacings.large)
+        override val extraLarge = OdsToken(OdsToken.Spacing.ExtraLarge, this@OdsSpacings.extraLarge)
+        override val extraExtraLarge = OdsToken(OdsToken.Spacing.ExtraExtraLarge, this@OdsSpacings.extraExtraLarge)
+    }
+}
+
+interface OdsSpacingsCatalog<T> : OdsThemeConfigurationItem.Catalog<T> {
```
Comment (paulinea) [inline theme-contract/src/main/java/com/orange/ods/theme/spacing/OdsSpacings.kt:44]: Here we have something generic but can we have something else than an `OdsSpacingCatalog` of `OdsSpacing`?

File: app/src/main/java/com/orange/ods/app/ui/components/chips/Chip.kt [hunk 2]
```diff
@@ -62,7 +62,7 @@ fun Chip(variant: Variant) {
                     Subtitle(textRes = R.string.component_element_leading, horizontalPadding = true)
                     OdsChoiceChipsFlowRow(
                         selectedChoiceChipIndex = ChipCustomizationState.LeadingElement.entries.indexOf(leadingElement.value),
-                        modifier = Modifier.padding(horizontal = dimensionResource(id = com.orange.ods.R.dimen.spacing_m)),
+                        modifier = Modifier.padding(horizontal = OdsTheme.spacings.medium.dp),
```
Comment (paulinea) [inline app/src/main/java/com/orange/ods/app/ui/components/chips/Chip.kt]: There is always a reference to `com.orange.ods.R.dimen.spacing_m` in `Chips.md` line 205

File: theme-contract/src/main/java/com/orange/ods/theme/spacing/OdsSpacing.kt [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    abstract val dp: Dp
+
+    /**
+     * An [OdsSpacing] resource value.
+     */
+    data class Resource(@DimenRes val id: Int) : OdsSpacing() {
+
+        override val dp: Dp
+            @Composable
+            get() = dimensionResource(id = id)
```
Comment (paulinea) [inline theme-contract/src/main/java/com/orange/ods/theme/spacing/OdsSpacing.kt]: As discussed together, we can remove `dimens` and simplify usage of spacings and remove `.dp` to access spacing value.

File: app/src/main/java/com/orange/ods/app/ui/guidelines/spacing/GuidelineSpacingScreen.kt [hunk 4]
```diff
@@ -57,19 +59,19 @@ fun GuidelineSpacingScreen() {
...
-        items(Spacing.entries) { spacing ->
-            val dp = spacing.getDp()
-            val ratio = spacing.getRatio()
+        items(spacings.tokens.entries) { spacing ->
+            val dp = spacing.value.dp
+            val ratio = dp / spacings.small.dp
             OdsListItem(
-                text = spacing.tokenName,
+                text = spacing.name,
                 secondaryText = stringResource(id = R.string.guideline_spacing_dp, dp.value.toInt()) + "\n",
```
Comment (paulinea) [inline app/src/main/java/com/orange/ods/app/ui/guidelines/spacing/GuidelineSpacingScreen.kt]: Do we need to display kotlin code for each item to use the spacing? `OdsTheme.spacings.small` for example.

**Review by paulinea — CHANGES_REQUESTED** [review]

File: theme-contract/src/main/java/com/orange/ods/theme/spacing/OdsSpacings.kt (hunk 1 above)
Comment (florentmaitre) [reply theme-contract/src/main/java/com/orange/ods/theme/spacing/OdsSpacings.kt:44]: Yes, we also have an `OdsSpacingsCatalog` of `OdsToken<OdsSpacing>`.

**Review by paulinea — APPROVED** [review]

**Pull request merged**